app/models/user.rb

```diff
@@ -3,6 +3,8 @@
 class User < ApplicationRecord
   require "pg_search"
   include SimpleDiscussion::ForumUser
+  validates :name, presence: true
+  validates :email, format: { with: /\b[A-Z0-9._%a-z\-]+@(?:[A-Z0-9a-z\-]+\.)+[A-Za-z]{2,4}\z/ }
```

Comment:
For consistency, is it possible to use the regex expression used by devise? or update the regex expression in devise.rb and reference it here.  

https://github.com/CircuitVerse/CircuitVerse/blob/75462c8b6c16b40d7556d53d379f3882f2ce5983/config/initializers/devise.rb#L176

Reply:
current devise regex provides simple validation and do not validates if domain is present or not 

to  just add domain name validation we can use \A[^@\s]+@([^@.\s]+\.)+[^@.\s]+\z  but it lacks other features 

\b[A-Z0-9+_%a-z\-"]+@(?:[A-Z0-9a-z]+\.)+[A-Za-z]{2,4}\z  this will be good as it also apply other basic validations 
I have tested  above both regex on https://rubular.com/  
shall I update this regex \b[A-Z0-9+_%a-z\-"]+@(?:[A-Z0-9a-z]+\.)+[A-Za-z]{2,4}\z   with the regex present in devise.rb?
this regex accepts and rejects valid and invalid email addresses  respectively as given in list in  following url
https://gist.github.com/cjaoude/fd9910626629b53c4d25